### DISCUSSION TITLE
Separate repo for languages

### DISCUSSION THREAD

**Discussion by niksacdev** [microsoft/semantic-kernel#1328] [Ideas]
Currently there is a mono repo for both python and dotnet, this creates an issue of dependencies where a user needs to install dotnet 7.0 (for Polyglot Notebooks and C# code) even when they only want to use Python. Can we create separate repos for each supported language and then create a wrapper repo that links to the underlying language repo. It can also help improve CI/CD performance as the repo size grows

Today
|-- All Source --}

Future
|---- Semantic-Kernel Main Repo--|
--- Overview 
---- Common Docs
---- Link to language repos
|---- SK - Dotnet ---|
---- Code
---- Sample
---- Docs
|---- SK - Python ---|
---- Code
---- Sample
---- Docs

**Comment by shawncal** [comment]
Hey @niksacdev, yeah I hear you. As the project grows, so does the PR list, issues, Actions prechecks on PRs... I predict that splitting may be in our near future. Thanks for the suggestion.

**Comment by kaza** [comment]
in Addtion the directory structure is a bit cumbersome which could be simplified:
samples/dotnet
samples/python
samples/notebooks
samples/notebooks/dotnet/
